frontend/src/hooks/useTyping.ts

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+
+export const useTyping = (text: string) => {
+  const [message, setMessage] = React.useState(text[0]);
+
+  const advance = () =>
+    setTimeout(() => {
+      if (message.length < text.length) {
+        setMessage(text.slice(0, message.length + 1));
+      }
+    }, 10);
+
+  React.useEffect(() => {
+    const timeout = advance();
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [message]);
```

Comment:
can we fix this missing dependency before merging? I'm kinda annoyed by the warnings below every PR and don't want there to be even more.